lib/frames.js

```diff
@@ -26,17 +42,253 @@ frames.writeFrame = function(frame, stream) {
 
   var performative = frame.toDescribedType();
   if (performative !== null && performative !== undefined) {
-    debug('Encoding performative: ' + JSON.stringify(performative));
     codec.encode(performative, builder);
-
-    if (frame instanceof TransferFrame) {
-      if (frame.message !== undefined) builder.appendBuffer(frame.message);
+    if (frame instanceof frames.TransferFrame) {
+      if (frame.payload !== undefined) builder.appendBuffer(frame.payload);
     }
   }
 
   var buffer = builder.get();
   buffer.writeUInt32BE(buffer.length, 0);
-
-  debug('sending frame: ' + frame.constructor.name + ': ' + JSON.stringify(buffer) + ': ' + buffer.toString('hex'));
+  debug('sending frame: ', frame, ': ' + buffer.toString('hex'));
   stream.write(buffer);
 };
+
+frames.readFrame = function(buffer) {
+  if (buffer.length < 8) return undefined;
+
+  var sizeAndDoff = buffer.slice(0, 8);
+  var size = sizeAndDoff.readUInt32BE(0);
+  if (size > buffer.length) return undefined;
+  buffer.consume(8);
+
+  var doff = sizeAndDoff[4];
+  var frameType = sizeAndDoff[5];
+  if (frameType !== FrameType.AMQP && frameType !== FrameType.SASL) {
```

Comment:
I almost feel like we should move this below `buffer.consume(payloadSize)` - that way if we just get some unknown frame type, we still error, but they could potentially swallow that and continue receiving on the next frame.